src/storage/src/hummock/compactor/mod.rs

```diff
@@ -359,21 +522,11 @@ pub fn start_compactor(
 
                 let running_task_parallelism = running_task_parallelism.clone();
                 let request_sender = request_sender.clone();
-                let event: Option<Result<SubscribeCompactionEventResponse, _>> = tokio::select! {
+                let event: Option<Result<CompactionEventResponses, _>> = tokio::select! {
```

Comment:
I'm still considering whether we should use the same loop to dispatch both Iceberg and Hummock logic. The reasons are: the workflows for these two services might be completely different. For example:  
1. Heartbeat behavior  
2. Tick intervals  
3. Whether to execute reporting  
4. Runner behavior  
5. Resource estimation and limits  
...and so on.


Of course, the above-mentioned refactoring does not need to be implemented in a single PR. The current workaround is already sufficient to support the ver 0.1 Iceberg compactor service.

 @hzxa21 Any suggestion?

Reply:
"I think executing them together is acceptable. Apart from some differences in specific computational implementations, the logic layers are consistent. What Hummock has, Iceberg should also have in the future.

Reply:
Considering the iceberg comapctor just handling pull task event, let's make this logic decoupled with the hummock at the very beginning. It would be much clearer and easier to evolve.